common/src/main/java/com/wynntils/wc/custom/item/WynnItemStack.java

```diff
@@ -35,19 +36,37 @@ public List<Component> getOriginalTooltip() {
     }
 
     public void addProperty(ItemProperty property) {
-        if (getProperty(property.getClass()) != null) return; // don't allow duplicate properties of same type
+        if (getProperty(property.getClass()) != null) return; // don't allow duplicate properties
         this.properties.add(property);
     }
 
-    public <T extends ItemProperty> T getProperty(Class<T> propertyType) {
+    /**
+     * Returns the specified property or the first property of the specified type, if it exists.
```

Comment:
It's not so much an "or" here, it's really just "return the first property of the specified type", is it not?

(Same goes for hasProperty)

Reply:
no, because it can take a specific `ItemProperty` or a `PropertyType`, and the latter will find the first property present that implements that type

Reply:
But you are scoping T to extend ItemProperty? How can you then send in a PropertyType?

Maybe this is all obvious to y'all, but I'm getting lost in the type hierarchy and casting here. 

What is the purpose of the PropertyType?

Reply:
I think its suppose to represent properties with multiple implementations.

Reply:
The `ItemProperty` scope is removed in this PR, so the method can accept both

`PropertyType` is for properties with different implementations that accomplish the same thing. In this PR specifically, regular gear items and cosmetic items differ far too much to handle with the same `ItemProperty` class, but the end goal is the same: draw a highlight behind the item. Thus, the `HighlightProperty` type exists so that the highlight rendering code doesn't have to define separate handling for every applicable `ItemProperty`

Reply:
Right, you remove the scope. I misread the diff.

It's still a bit messy in my head, but I think we can blame that more on me than your code. 😉

Reply:
it's certainly a bit convoluted, but due to the nature of what we're doing here i think it's either this, or lots and lots of `if` statements (legacy wynntils' approach). i'll try to add some documentation in the repo wiki later so everyone can get a clear picture of it